public/app/features/explore/state/query.ts

```diff
@@ -320,29 +314,14 @@ async function handleHistory(
 ) {
   const datasourceId = datasource.meta.id;
   const nextHistory = updateHistory(history, datasourceId, queries);
-  const {
-    richHistory: nextRichHistory,
-    richHistoryStorageFull,
-    limitExceeded,
-  } = await addToRichHistory(
-    state.richHistory || [],
-    datasource.uid,
-    datasource.name,
-    queries,
-    false,
-    '',
-    !state.richHistoryStorageFull,
-    !state.richHistoryLimitExceededWarningShown
-  );
   dispatch(historyUpdatedAction({ exploreId, history: nextHistory }));
-  dispatch(richHistoryUpdatedAction({ richHistory: nextRichHistory }));
 
-  if (richHistoryStorageFull) {
-    dispatch(richHistoryStorageFullAction());
-  }
-  if (limitExceeded) {
-    dispatch(richHistoryLimitExceededAction());
-  }
+  dispatch(addHistoryItem(datasource.uid, datasource.name, queries));
+
+  // Because filtering happens in the backend we cannot add a new entry without checking if it matches currently
+  // used filters. Instead, we refresh the query history list.
+  // TODO: run only if Query History list is opened
```

Comment:
is this intended to be left for a future PR?

Reply:
Yes, it should be done as part of #47252. I've added issue number to the TODO in 14e3b73c4757c6318ab1e63627d51696497b0ee1